pkg/accounting/accounting.go

```diff
@@ -216,6 +221,65 @@ func (a *Accounting) getPeerBalance(peer swarm.Address) (*PeerBalance, error) {
 	return peerBalance, nil
 }
 
+func (a *Accounting) Balances() (map[string]int64, error) {
+	peersBalances := make(map[string]int64)
+
+	a.balancesMu.Lock()
+	for peer, balance := range a.balances {
+		peersBalances[peer] = balance.balance
+	}
+	a.balancesMu.Unlock()
+
+	err := a.CompleteFromStore(peersBalances)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return peersBalances, nil
+}
+
+func (a *Accounting) CompleteFromStore(s map[string]int64) error {
+
+	err := a.store.Iterate(balancesPrefix, func(key, val []byte) (stop bool, err error) {
+
+		addr := balanceKeyPeer(key)
+
+		if _, ok := s[addr.String()]; !ok {
+			var storevalue int64
+			err = a.store.Get(peerBalanceKey(addr), &storevalue)
+			if err != nil {
+				a.logger.Debugf("store operation error for peer %v: %v", addr.String(), err)
+				return false, nil
+			}
+			s[addr.String()] = storevalue
+		}
+
+		return false, nil
+	})
+
+	return err
+
+}
+
+func balanceKeyPeer(key []byte) swarm.Address {
+	k := string(key)
+	var ret swarm.Address
```

Comment:
this is not necessary to allocate.
you can remove this line and in the cases below where you return `ret` just `return swarm.ZeroAddress` instead. this is not contradiction even if you change the signature to `Address,error`, since you would still need to return some value for the address when returning an error